src/halfpipe/tui/working_directory/base.py

```diff
@@ -0,0 +1,525 @@
+        """
+        Loads settings from the 'spec.json' file into the context cache.
+
+        This method reads the settings from the 'spec.json' file (if it
+        exists) and populates the context cache with the loaded data.
+        This includes global settings, file patterns, feature settings,
+        and model settings.
+        """
+        # First feed the cache
+        if self.existing_spec is not None:  # Add this check
+            global_settings = self.existing_spec.global_settings
+            files = self.existing_spec.files
+            settings = self.existing_spec.settings
+            features = self.existing_spec.features
+            models = self.existing_spec.models
+
+            ctx.spec.global_settings = global_settings
+            for fileobj in files:
+                ctx.put(fileobj)
+            ctx.spec.settings = settings
+            ctx.spec.features = features
+            ctx.spec.models = models
+
+            self.cache_file_patterns()
+
+    @work(exclusive=True, name="cache_file_worker")
+    async def cache_file_patterns(self):
+        """
+        Caches file patterns and settings from the spec file.
+
+        This method is a worker that processes the loaded specification
+        and caches the file patterns and settings in the context. It also
+        creates the corresponding widgets for data input, preprocessing,
+        feature selection, and model selection.
+        """
+        data_input_widget = self.app.get_widget_by_id("input_data_content")
+        preprocessing_widget = self.app.get_widget_by_id("preprocessing_content")
+        self.feature_widget = self.app.get_widget_by_id("feature_selection_content")
+        self.model_widget = self.app.get_widget_by_id("models_content")
+
+        # The philosophie here is that we copy the data from the existing spec file to the context cache and then create
+        # corresponding widgets. Through these widgets then it should be possible to further modify the spec file.
+        # The created widgets should avoid using step and meta classes upon creation as these triggers various user choice
+        # modals.
+        if self.existing_spec is not None:
+            self.app.get_widget_by_id("input_data_content").toggle_bids_non_bids_format(False)
+            spreadsheet_counter = 0
+            self.event_file_objects = []
+            self.atlas_file_objects = []
+            self.seed_map_file_objects = []
+            self.spatial_map_file_objects = []
+
+            preprocessing_widget.default_settings = self.existing_spec.global_settings
+            preprocessing_widget = preprocessing_widget.refresh(recompose=True)
+
+            for f in self.existing_spec.files:
+                # In the spec file, subject is abbreviated to 'sub' and atlas, seed and map is replaced by desc,
+                # here we replace it back for the consistency.
+                f.__dict__["path"] = f.__dict__["path"].replace("{sub}", "{subject}")
```

Comment:
Could you do this in a loop with the `entity_longnames` dictionary, just to be safe? https://github.com/HALFpipe/HALFpipe/blob/430f383486fd5fba07e98b0c10a73346a188b65c/src/halfpipe/model/tags/schema.py#L33

Reply:
yes